src/hotspot/share/opto/callnode.hpp

```diff
@@ -769,6 +770,27 @@ class CallDynamicJavaNode : public CallJavaNode {
 #endif
 };
 
+//------------------------------CallBlackholeNode----------------------------
+// Make a blackholed call. It would survive through the compiler and keep
+// the effects on its argument, and would be finally emitted as nothing.
+class CallBlackholeNode : public CallNode {
+  virtual bool cmp( const Node &n ) const;
+  virtual uint size_of() const; // Size is bigger
+public:
+  CallBlackholeNode( const TypeFunc *tf , address addr) : CallNode(tf, addr, TypePtr::BOTTOM) {
```

Comment:
`addr` can be omitted since it is always `NULL`.

Reply:
Right, fixed.